### DISCUSSION TITLE
Is not a function

### DISCUSSION THREAD

**Discussion by maarten-mald-digital** [CodeDredd/pinia-orm#1913] [Q&A] [answered]
Methods from my Models not working. I am on Nuxt 3 and using the comp API. 

models/Project.ts 

```
import { Model } from 'pinia-orm';
import Sprint from '@/models/Sprint';
import Task from '@/models/Task';

export default class Project extends Model {
	static entity = 'projects';

	/**
	 * Extra options for persisting to localStorage
	 */
	static piniaOptions = {
		persist: true,
	};

	/**
	 * List of all fields and relationships
	 */
	static fields() {
		return {
			id: this.uid(),
			title: this.string(''),
			sprints: this.hasMany(Sprint, 'project_id', 'id'),
			tasks: this.hasMany(Task, 'project_id', 'id'),
			created_at: this.string(null),
			updated_at: this.string(null),
		};
	}

	get myTest(): string {
		return 'test';
	}

	isMotion(): string {
		return 'test';
	}

	// For typescript support of the field include also the next lines
	declare id: number;
	declare title: string;
	declare created_at: string;
	declare updated_at: string;
}
```

in my component edit.vue

```
<script lang="ts" setup>
import Project from '@/models/Project';
import { ProjectRepository } from '@/repositories/ProjectRepository';

const { id: projectId } = useRoute().params;
const project = computed(() => useRepo(Project).find(projectId));

onMounted(() => {
	setTimeout(() => {
		console.log('mounted');
		console.log(project.isMotion);
		console.log(project.isMotion());

		console.log(project.myTest);
		console.log(project.myTest());
	}, 2000);
});
</script>
```

isMotion/isMotion()/myTest/myTest are not working.

![Screenshot 2024-07-28 at 17 43 08](https://github.com/user-attachments/assets/59a6e482-152f-4b47-84ea-a273797e42c1)

My feeling is that it is not recognized as a "Project" model. I checked the docs/chatgpt but couldnt solve it.

**Comment by CodeDredd** [comment]
Hi,

i think you need to define your function as static to work that way. Have you tried out?

Best,
Gregor
________________________________
Von: maarten-mald-digital ***@***.***>
Gesendet: Sonntag, 28. Juli 2024 17:48
An: CodeDredd/pinia-orm ***@***.***>
Cc: Subscribed ***@***.***>
Betreff: [CodeDredd/pinia-orm] Is not a function (Discussion #1913)


Methods from my Models not working. I am on Nuxt 3 and using the comp API.

In my models/Project.ts

`
get myTest(): string {
return 'test';
}

isMotion(): string {
return 'test';
}
`

in my component edit.vue

`

<script lang="ts" setup> import Project from '@/models/Project'; import { ProjectRepository } from '@/repositories/ProjectRepository'; const { id: projectId } = useRoute().params; const project = computed(() => useRepo(Project).find(projectId)); onMounted(() => { setTimeout(() => { console.log('mounted'); console.log(project.isMotion); console.log(project.isMotion()); console.log(project.myTest); console.log(project.myTest()); }, 2000); }); </script>

`

isMotion/isMotion()/myTest/myTest are not working.

Screenshot.2024-07-28.at.17.43.08.png (view on web)<https://github.com/user-attachments/assets/59a6e482-152f-4b47-84ea-a273797e42c1>

My feeling is that it is not recognized as a "Project" model. I checked the docs/chatgpt but couldnt solve it.

**Reply by maarten-mald-digital** [reply, marked as answer]
defining as static only works when i am using Project (as a class).

My mistake was not using .value (composition api). So project.value.isMotion();

Thanks for the fast reply!

**Reply by CodeDredd** [reply]
ahhhh true
